tests/unit/test_charm.py

```diff
@@ -77,8 +65,71 @@ def test__on_config_changed(harness: ops.testing.Harness, monkeypatch: pytest.Mo
     get_relation_mock = MagicMock()
     monkeypatch.setattr(ops.Model, "get_relation", get_relation_mock)
 
-    jenkins_charm: JenkinsAgentCharm = harness.charm
-    jenkins_charm._on_config_changed(config_changed_event)
+    charm: JenkinsAgentCharm = harness.charm
+    charm._on_config_changed(config_changed_event)
```

Comment:
Is is possible to trigger a config changed to test this instead of executing the handler?

Reply:
Yes, I refactor most of the tests and removed the pytest exception together with it